### PR TITLE
Adding telemetry publish for unhandled exceptions to developer exception page

### PR DESCRIPTION
**Pull request by JunTaoLuo**
For #180 cc @Eilon @Tratcher

### CODE REVIEW

File: test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs [hunk 1]
```diff
@@ -471,5 +477,72 @@ public class ErrorData
...
+            telemetryListener.SubscribeWithAdapter(listener);
+
+            // Act
+            await server.CreateClient().GetAsync("/path");
+            
+            // Assert
+            Assert.NotNull(listener.DiagnosticUnhandledException?.HttpContext);
+            Assert.NotNull(listener.DiagnosticUnhandledException?.Exception);
+            Assert.Null(listener.HostingUnhandledException?.HttpContext);
+            Assert.Null(listener.HostingUnhandledException?.Exception);
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs]: Assert NotNull HostingEndRequest

Comment (JunTaoLuo) [inline test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs]: I'm guessing I should Assert NotNull HostingBeginRequest as well?

**Comment by Tratcher** [comment]
@avanderhoorn @rynowak

**Comment by Tratcher** [comment]
Include the bug number in the commit message.

File: test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs (hunk 1 above)
Comment (Tratcher) [inline test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs]: *shrug* that's out of scope of this change.

**Comment by JunTaoLuo** [comment]
@Tratcher I will add the bug number once I squash all commits

**Comment by avanderhoorn** [comment]
When is this event triggered as apposed to `"Microsoft.AspNet.Hosting.UnhandledException"` [here](https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs#L119).

**Comment by JunTaoLuo** [comment]
@avanderhoorn When the developer exception page successfully displays. Since the exception will be suppressed, we want to publish it to telemetry source as diagnostic unhandled exception. If for other reasons displaying the exception fails, the exception will be bubbled up to hosting and will trigger the telemetry publish there.

**Comment by avanderhoorn** [comment]
Great, so in the case that this event happens, we wont get the `hosting` one.

**Comment by JunTaoLuo** [comment]
@avanderhoorn right.

**Comment by avanderhoorn** [comment]
Makes sense. Thanks! :shipit:

**Comment by Tratcher** [comment]
We also need this here: https://github.com/aspnet/Diagnostics/blob/dev/src/Microsoft.AspNet.Diagnostics/ExceptionHandler/ExceptionHandlerMiddleware.cs#L66

**Comment by JunTaoLuo** [comment]
@Tratcher I was asking for that in my previous comment in the issue #180 . That error is technically handled and I wasn't sure if that should also publish to telemetry.

**Comment by avanderhoorn** [comment]
For our part we would like to know about that exception too.

**Comment by JunTaoLuo** [comment]
@avanderhoorn cool. I added the telemetry publish for exception handler as well. It's going to publish to `Microsoft.AspNet.Diagnostics.HandledException` just to differentiate from the unhandled exception from developer exception page, is that good?

**Comment by avanderhoorn** [comment]
Sounds great!

**Comment by JunTaoLuo** [comment]
I'm going to take that as a :shipit: All tests passed so I'll merge to dev.

**Comment by avanderhoorn** [comment]
Thanks!

**Pull request merged**